acra/configuring-maintaining/general-configuration/acra_keymaker.md

```diff
@@ -0,0 +1,178 @@
+---
+title: AcraKeymaker
+bookCollapseSection: true
+---
+
+# AcraKeymaker
+
+## Command line flags
+
+### General flags
+
+* `--client_id=<id>`
+  
+  Use provided client ID as identifier to generate keys or keypairs.
+
+* `--tls_cert=<path>`
+  
+  Path to TLS certificate to use as client_id identifier. Works only when passed empty ClientID: `--client_id=""`.
+
+* `--tls_identifier_extractor_type={distinguished_name|serial_number}`
+  
+  Decide which field of TLS certificate to use as ClientID.
+
+  * `distinguished_name` — (default) certificate Distinguished Name (DN).
+  * `serial_number` — certificate serial number.
+
+* `--keystore=<v1|v2>`
+  
+  Set keystore format. If `--keys_output_dir` points to existing keystore destination then `acra-keymaker` automatically detects version of keystore.
+  If destination storage is empty and hasn't any key then this argument is required. Read more about [keystore versions]({{< ref "/acra/keys/versions/" >}})
+  
+### Storage destination
+
+#### Filesystem
+
+* `--keys_output_dir=<folder>`
+  
+  Folder where will be saved keys (you can change destination for asymmetric public keys with keystore V1 (V2 store at same destination)  with parameter `--keys_public_output_dir`).  
+  Default is `.acrakeys`.
+
+* `--keys_public_output_dir=<folder>`
+  
+  Folder where will be saved public asymmetric keys. Works only for `--keystore=v1` and do nothing with `--keystore=v2`.
+  Default is `.acrakeys`.
```

Comment:
You need to pick either “directory” or ”folder” and stick with it. (Also, you can keep things more simple. (Also, there are (too) many (nested) parentheses here, (I think).))

```suggestion
* `--keys_output_dir=<path>`

  Path to keystore directory.

  Default is `.acrakeys`.

* `--keys_public_output_dir=<folder>`

  Path to public key directory for `--keystore=v1` format.
  (Ignored for `--keystore=v2`.)

  Default is `.acrakeys`.
```